src/main/java/com/minecolonies/coremod/util/StructureWrapper.java

```diff
@@ -354,26 +395,27 @@ public boolean isStructureBlockEqualWorldBlock()
         {
             return structureBlock == worldBlock;
         }
-        else if(worldBlock == ModBlocks.blockRack)
+        else if (worldBlock == ModBlocks.blockRack)
         {
             return BlockMinecoloniesRack.shouldBlockBeReplacedWithRack(structureBlock);
         }
         else if ((structureBlock instanceof BlockStairs && structureBlockState == worldBlockState)
-                || BlockUtils.isGrassOrDirt(structureBlock, worldBlock, structureBlockState, worldBlockState))
+                || BlockUtils.isGrassOrDirt(structureBlock, worldBlock, structureBlockState, worldBlockState)
+                || structureBlock instanceof BlockWaypoint)
         {
             return true;
         }
 
         final Template.EntityInfo entityInfo = structure.getEntityinfo(this.getLocalPosition());
-        if(entityInfo != null)
+        if (entityInfo != null)
         {
             return false;
             //todo get entity at position.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)